src/modules/visitor-space/components/CreatedFilterForm/CreatedFilterForm.tsx

```diff
@@ -74,84 +84,160 @@ const CreatedFilterForm: FC<CreatedFilterFormProps> = ({ children, className, di
-											?.value as Operator;
-
-										if (value !== form.operator) {
-											setForm({
-												operator: value,
-												created: defaultValues.created,
-											});
+					/>
+				</FormControl>
+			</div>
+			<div className={clsx(styles['createdFilterForm__date'], 'u-px-20 u-px-32:md')}>
+				<FormControl
+					className="u-mb-24 c-form-control--label-hidden"
+					errors={[errors.created?.message]}
+					id={labelKeys.created}
+					label={tHtml(
+						'modules/visitor-space/components/created-filter-form/created-filter-form___waarde'
+					)}
+				>
+					<SelectDateOrYear
+						yearsSelected={yearsSelected}
+						setYearsSelected={setYearsSelected}
+						showPluralLabel={showRange}
+					/>
+					<Controller
+						control={control}
+						name="created"
+						render={() =>
+							yearsSelected && showRange ? (
+								<YearRangeInput
+									disabled={disabled}
+									showLabels
+									id={labelKeys.created}
+									onChange={(e) => {
+										setYearRange(e.target.value);
+									}}
+									value={yearRange}
+								/>
+							) : showRange ? (
+								<DateRangeInput
+									disabled={disabled}
+									showLabels
+									id={labelKeys.created}
+									onChange={(e) => {
+										onChangeCreated(e.target.value);
+									}}
+									value={form.created}
+								/>
+							) : yearsSelected ? (
+								<YearInput
+									label={getSelectValue(operators, form.operator)?.label}
+									disabled={disabled}
+									id={labelKeys.created}
+									onChange={(e) => {
+										if (e.target.value.length < 5) {
+											setYear(e.target.value);
 										}
 									}}
-									options={operators}
-									value={getSelectValue(operators, field.value)}
+									value={year}
 								/>
-							)}
-						/>
-					</FormControl>
-
-					<FormControl
-						className="u-mb-24 c-form-control--label-hidden"
-						errors={[errors.created?.message]}
-						id={labelKeys.created}
-						label={tHtml(
-							'modules/visitor-space/components/created-filter-form/created-filter-form___waarde'
-						)}
-					>
-						<Controller
-							control={control}
-							name="created"
-							render={() =>
-								showRange ? (
-									<DateRangeInput
-										disabled={disabled}
-										id={labelKeys.created}
-										onChange={(e) => onChangeCreated(e.target.value)}
-										value={form.created}
-									/>
-								) : (
-									<DateInput
-										disabled={disabled}
-										id={labelKeys.created}
-										onChange={(date) => {
-											onChangeCreated(
-												(date || new Date()).valueOf().toString()
-											);
-										}}
-										value={form.created}
-									/>
-								)
-							}
-						/>
-					</FormControl>
-				</div>
+							) : (
+								<DateInput
+									label={getSelectValue(operators, form.operator)?.label}
+									disabled={disabled}
+									id={labelKeys.created}
+									onChange={(date) => {
```

Comment:
Maybe move this to a function above the render e.g. `onChangeDateInputChange`. This would make the render fn easier to read. Same for the other onChange functions.